local_planner/src/nodes/local_planner.cpp

```diff
@@ -169,97 +160,42 @@ void LocalPlanner::determineStrategy() {
       waypoint_type_ = direct;
     }
 
-    if (send_obstacles_fcu_) {
-      create2DObstacleRepresentation(true);
-    }
-  } else if (final_cloud_.points.size() > min_cloud_size_ && stop_in_front_ &&
-             reach_altitude_) {
-    obstacle_ = true;
-    ROS_INFO(
-        "\033[1;35m[OA] There is an Obstacle Ahead stop in front\n \033[0m");
-    stopInFrontObstacles();
-    waypoint_type_ = direct;
-
     if (send_obstacles_fcu_) {
       create2DObstacleRepresentation(true);
     }
   } else {
-    if (((counter_close_points_backoff_ > 200 &&
-          final_cloud_.points.size() > min_cloud_size_) ||
-         back_off_) &&
-        reach_altitude_ && use_back_off_) {
-      if (!back_off_) {
-        back_off_point_ = closest_point_;
-        back_off_start_point_ = position_;
-        back_off_ = true;
-      } else {
-        float dist = (position_ - back_off_point_).norm();
-        if (dist > min_dist_backoff_ + 1.0f) {
-          back_off_ = false;
-        }
-      }
-      waypoint_type_ = goBack;
-      if (send_obstacles_fcu_) {
-        create2DObstacleRepresentation(true);
-      }
+    waypoint_type_ = tryPath;
+
+    evaluateProgressRate();
+    create2DObstacleRepresentation(send_obstacles_fcu_);
 
+    // decide how to proceed
+    if (hist_is_empty_) {
+      obstacle_ = false;
     } else {
-      evaluateProgressRate();
+      obstacle_ = true;
```

Comment:
It looks like `obstacle_` and `hist_is_emtpy_` essentially contain the same information. Can we get rid of one of them?